ansible_base/authentication/utils/claims.py

```diff
@@ -195,9 +188,9 @@ def has_access_with_join(current_access: Optional[bool], new_access: bool, condi
         return current_access and new_access
 
 
-def process_user_attributes(trigger_condition: dict, attributes: dict, authenticator_id: int) -> Optional[bool]:
+def process_user_attributes(trigger_condition: dict, attributes: dict, authenticator_id: int) -> Optional[Literal[True]]:
```

Comment:
yes, that's a core of this change. If these methods return False, later processing can't distinguish if the `revoke` flag was used. 
By returning `None` with `revoke: False` means that the authentication map is ignored (and should be ignored)

Reply:
This is a requested change during review, so there is conflict in reviewers. 
So either there will be 3rd reviewer, or I vote for the current status, I find it useful - it's saying it *can't* return False

Reply:
The enum is an interesting option rather than True/False/None. I see where Literal could be being misused by my suggestion; but it does, to me, feel misleading to say `Optional[bool]`.  The function is currently intended to process the trigger itself which should be `True or None` the `False` comes into play later when looking at the results of the trigger and a property on the map. So I don't see this ever returning `False` unless we change the argument from the trigger to the map itself.

Given that I could go in one of two directions.
1. Leave the Optional[Literal[True]] - maybe suboptimal given the link above.
2. Defe some kind of enum or constant that would only contain ALLOW or SKIP (not DENY) and return one of those.

Reply:
> If the requirements change again, the function may start return False again

I don' t understand this argument - you can never define anything with this. requirements of everything can change in the future, so with this logic you can't write any type checking (or even any method definition without `**kwargs` etc.)

Reply:
to the enum: as a developer reading the code without knowledge what is going on inside, I'll be totally lost with the Enum. It's conterproductive to document method, which can return True or None with the enum describing in can return `auto()` or `auto()`. I (and I'm convinced most people) won't understand it. 
So either let's go back to `Optional(bool)` or keep it as is.

Reply:
clarified during meeting